aspnetcore-webapi/UsingResultPatternInNETWebAPI/UsingResultPatternInNETWebAPI/Repositories/IContactsRepository.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace UsingResultPatternInNETWebAPI.Repositories;
+
+public interface IContactsRepository
+{
+    IEnumerable<Contact> GetAll();
+    Contact? GetById(Guid id);
+    Contact Create(Contact contact);
+    bool ExistsByEmail(string email);
```

Comment:
Instead of ExistsByEmail method, I would prefer to have GetByEmail method, the way we have GetById

Reply:
> Instead of ExistsByEmail method, I would prefer to have GetByEmail method, the way we have GetById

GetById would suggest that we are returning the contact, while this method is only to verify that this contact exists, are you sure that we should that?

Reply:
This is just a sample project and it is not the core part which we are focusing in the article, so it hardly matters. But why I suggested this approach is that, if in the real application we need to fetch the record using EmailId or want to check if the record exists with EmailId, with a single method we can cover both the scenario.